src/Workspaces/CSharp/Portable/Execution/CSharpOptionsSerializationService.cs

```diff
@@ -84,8 +84,12 @@ public override ParseOptions ReadParseOptionsFrom(ObjectReader reader, Cancellat
             var languageVersion = (LanguageVersion)reader.ReadInt32();
             var preprocessorSymbolNames = reader.ReadArray<string>();
 
-            var options = new CSharpParseOptions(languageVersion, documentationMode, kind, preprocessorSymbolNames);
-            return options.WithFeatures(features);
+            var options = new CSharpParseOptions(languageVersion, documentationMode, kind);
+
+            // use WithPreprocessorSymbols instead of constructor to bypass preprocessor validation.
+            // https://github.com/dotnet/roslyn/issues/15797
+            return options.WithPreprocessorSymbols(preprocessorSymbolNames)
```

Comment:
looks like we have been this way for a while.

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices.CSharp/ProjectSystemShim/CSharpProjectShim.cs,247

OOP is NOT doing any behavior differently than inproc Roslyn ProjectShim is doing.